src/libcore/mem.rs

```diff
@@ -448,17 +448,29 @@ pub unsafe fn uninitialized<T>() -> T {
 pub fn swap<T>(x: &mut T, y: &mut T) {
     unsafe {
         // Give ourselves some scratch space to work with
-        let mut t: T = uninitialized();
+        let mut t: [u8; 16] = uninitialized();
```

Comment:
for reference, the actual block size didn't matter too much but that was just on my machine - 16 may very well not be optimal